### PR TITLE
Update crypto.md

### PR DESCRIPTION
**Pull request by vin01**
Added length extension attack. It is pretty common for file downloads, redirection validation tokens and such.

### CODE REVIEW

**Comment by EdOverflow** [comment]
Thank you, @vin01. The only issue that I have with this is the fact that this is not a "universal" payload. It is purely a reference. One cannot simply copy-paste this payload in order to perform a hash length extension attack. So maybe adding a bit more info about what the attacker has to know in order to perform this type of attack might help.

**Comment by vin01** [comment]
Hey, In this case, the payload will vary based on the type of hash and usage. Other than the blog post with a good detailed explanation, I have linked an H1 report and the tool which may be used to extend a hash and exploit such a vulnerability.

File: cheatsheets/crypto.md [hunk 1]
```diff
@@ -47,3 +47,31 @@
 ```
 0123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345
 ```
+
+**Length extension attack**
+
+In cryptography and computer security, a length extension attack is a type of attack where an attacker can use `Hash(message1)` and the length of message1 to calculate `Hash(message1 ∥ message2)` for an attacker-controlled message2.
```
Comment (EdOverflow) [inline cheatsheets/crypto.md]: `message1` and `message2` should probably be enclosed in backticks:

> In cryptography and computer security, a length extension attack is a type of attack where an attacker can use `Hash(message1)` and the length of message1 to calculate `Hash(message1 ∥ message2)` for an attacker-controlled message2.

vs


> In cryptography and computer security, a length extension attack is a type of attack where an attacker can use `Hash(message1)` and the length of `message1` to calculate `Hash(message1 ∥ message2)` for an attacker-controlled `message2`.

File: cheatsheets/crypto.md [hunk 2]
```diff
@@ -47,3 +47,31 @@
...
+
+TL;DR: given a hash that is composed of a string with an unknown prefix, an attacker can append to the string and produce a new hash that still has the unknown prefix.
```
Comment (EdOverflow) [inline cheatsheets/crypto.md]: We should probably try to stay away from abbreviations that have not been defined earlier. Maybe add "In summary".

File: cheatsheets/crypto.md [hunk 3]
```diff
@@ -47,3 +47,31 @@
...
+```
+
+```
+http://example.com/download?file=report.pdf%80%00%00%00%00%00%00%00%00%00%00%00%00%00%00%00%00
+
+%00%00%00%00%00%00%00%00%00%00%00%00%00%00%00%00%00%00%00%00%00%00%00
+
+%00%00%A8/../../../../../../../etc/passwd&mac=ee40aa8ec0cfafb7e2ec4de20943b673968857a5
+```
+A related H1 report: https://hackerone.com/reports/251572
```
Comment (EdOverflow) [inline cheatsheets/crypto.md]: s/H1/HackerOne

**Comment by vin01** [comment]
Updated!

**Comment by EdOverflow** [comment]
Wonderful. Thank you, @vin01.

**Pull request merged**